solvers/pgd_solver.py

```diff
@@ -19,42 +19,57 @@ def set_objective(self, D, y, lmbd):
 
     # Main function of the solver, which computes a solution estimate.
     def run(self, callback):
+        """
+        PGD with FISTA
+
+        Parameters
+        ----------
+        callback : function
+            callback from benchopt
+        """
+        # Computes Lipschitz upper bound with FFT
         fourier_dico = fft(self.D, axis=1)
         L = np.max(np.real(fourier_dico * np.conj(fourier_dico)), axis=1).sum()
+
+        # Step size and regularization
         step_size = 1. / L
         mu = step_size * self.lmbd
 
         n_atoms = self.D.shape[0]
-        signal_length = self.y.shape[0]
+        signal_length = self.y.shape[1]
         kernel_size = self.D.shape[1]
-        n_samples = self.y.shape[1]
+        n_samples = self.y.shape[0]
```

Comment:
Use tuple expansion to get the shape of a numpy array
```suggestion
        n_samples, signal_length = self.y.shape
```